src/tests/ftest/cart/tests_common.h

```diff
@@ -498,4 +493,80 @@ tc_srv_start_basic(char *srv_group_name, crt_context_t *crt_ctx,
 	D_ASSERTF(rc == 0, "crt_group_size() failed; rc=%d\n", rc);
 }
 
+struct tc_log_msg_cb_resp {
+	sem_t	sem;
+	int	rc;
+};
+
+static void
+tc_log_msg_cb(const struct crt_cb_info *info)
+{
+	struct tc_log_msg_cb_resp	*resp;
+
+	if (info->cci_rc != 0) {
+		D_WARN(" Add Log message CB failure\n");
+		return;
+	}
+	resp = (struct tc_log_msg_cb_resp *)info->cci_arg;
+	resp->rc = info->cci_rc;
+	sem_post(&resp->sem);
+}
+
+int
+tc_log_msg(crt_context_t ctx, crt_group_t *grp, d_rank_t rank,
+	   char *msg) {
+	int32_t			     rc = 0;
+	struct crt_ctl_log_add_msg_in   *send_args;
+	struct crt_ctl_log_add_msg_out  *recv_args;
+	crt_rpc_t		       *rpc_req = NULL;
+	crt_endpoint_t			ep;
+	crt_opcode_t			opcode = CRT_OPC_CTL_LOG_ADD_MSG;
+	struct tc_log_msg_cb_resp       resp;
+
+	/* Initialize response structure */
+	rc = sem_init(&resp.sem, 0, 0);
+	D_ASSERTF(rc == 0, "sem_init() failed\n");
+	resp.rc = 0;
+
+	/* Fill in the endpoint info */
+	ep.ep_grp = grp;
+	ep.ep_rank = rank;
+	ep.ep_tag = 0;
+
+	rc = crt_req_create(ctx, &ep, opcode, &rpc_req);
+	if (rc != 0) {
+		D_ERROR("crt_req_create() failed. rc %d.\n", rc);
+		D_GOTO(endcode, rc);
+	}
+
+	crt_req_addref(rpc_req);
+	send_args =  crt_req_get(rpc_req);
+	send_args->log_msg = msg;
+
+	/* send the request */
+	rc = crt_req_send(rpc_req, tc_log_msg_cb, &resp);
+
```

Comment:
this call can fail in which case you should goto cleanup portion

Reply:
Added test for failure.